File: LLM-Translations/GPT-4/codenet/Java/C/s196249385.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void solve() {
    int n, k;
    scanf("%d %d", &n, &k);

    int *a = (int *)malloc(n * sizeof(int));
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }

    long *cumsum = (long *)malloc((n + 1) * sizeof(long));
    cumsum[0] = 0;
    for (int i = 0; i < n; i++) {
        cumsum[i + 1] = cumsum[i] + a[i];
    }

    long *b = (long *)malloc(n * (n + 1) / 2 * sizeof(long));
    int b_size = 0;
    long max = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            b[b_size++] = cumsum[j + 1] - cumsum[i];
            max = max > (cumsum[j + 1] - cumsum[i]) ? max : (cumsum[j + 1] - cumsum[i]);
        }
    }

    long mask = 1;
    while (mask < max) {
        mask <<= 1;
    }

    while (mask > 0) {
        long *tmp = (long *)malloc(n * (n + 1) / 2 * sizeof(long));
        int tmp_size = 0;
        for (int i = 0; i < b_size; i++) {
            if ((b[i] & mask) != 0) {
                tmp[tmp_size++] = b[i];
            }
        }

        if (tmp_size >= k) {
            memcpy(b, tmp, tmp_size * sizeof(long));
            b_size = tmp_size;
        }

        mask >>= 1;
        free(tmp);
    }

    if (b_size < k) {
        printf("0\n");
    } else {
        long ans = -1;
        for (int i = 0; i < b_size; i++) {
            ans &= b[i];
        }

        printf("%ld\n", ans);
    }

    free(a);
    free(b);
    free(cumsum);
}

int main() {
    solve();

    return 0;
}

// 

File: LLM-Translations/GPT-4/codenet/Java/Python/s196249385.py
import sys
from typing import List

def solve() -> None:
    n, k = map(int, input().split())
    a = list(map(int, input().split()))

    cumsum = [0] * (n + 1)
    for i in range(n):
        cumsum[i + 1] = cumsum[i] + a[i]

    b = []
    max_val = 0
    for i in range(n):
        for j in range(i, n):
            b.append(cumsum[j + 1] - cumsum[i])
            max_val = max(max_val, cumsum[j + 1] - cumsum[i])

    mask = 1 << (max_val.bit_length() - 1)
    while mask > 0:
        tmp = [e for e in b if e & mask != 0]
        
        if len(tmp) >= k:
            b = tmp
        
        mask >>= 1
    
    if len(b) < k:
        print(0)
    else:
        ans = -1
        for e in b:
            ans &= e
        
        print(ans)

def main() -> None:
    solve()

if __name__ == "__main__":
    main()

